numcodecs/shuffle.py

```diff
@@ -0,0 +1,109 @@
+
+# The _doShuffle and _doUnshuffle functions have been lifted from HSDS which
+# includes the following notice:
+##############################################################################
+# Copyright by The HDF Group.                                                #
+# All rights reserved.                                                       #
+#                                                                            #
+# This file is part of HSDS (HDF5 Scalable Data Service), Libraries and      #
+# Utilities.  The full HSDS copyright notice, including                      #
+# terms governing use, modification, and redistribution, is contained in     #
+# the file COPYING, which can be found at the root of the source code        #
+# distribution tree.  If you do not have access to this file, you may        #
+# request a copy from help@hdfgroup.org.                                     #
+##############################################################################
```

Comment:
Do we want to be lifting their code and including it here? Is there a way to implement this that does not rely on their code? Seems like this raises some licensing concerns

Reply:
Yep for sure, I just wanted to be transparent. Perhaps @jreadey could offer some guidance. I think that the shuffle algorithm would be considered public domain? In which case can we just rename some variables ;-)

Reply:
I think this can be resolved, the cython code is mostly rewritten compared to using numba